torch/csrc/api/include/torch/nn/functional/dropout.h

```diff
@@ -0,0 +1,19 @@
+#pragma once
+
+#include <torch/nn/options/dropout.h>
+
+namespace torch {
+namespace nn {
+namespace functional {
+
+using AlphaDropoutOptions = DropoutOptions;
+
+inline Tensor alpha_dropout(const Tensor& input, const AlphaDropoutOptions& options = {}, bool training = false) {
+  TORCH_CHECK(options.p() >= 0, "Dropout rate must not be less than zero");
+  TORCH_CHECK(options.p() <= 1, "Dropout rate must not be greater than one");
+  
+  return torch::alpha_dropout(input, options.p(), training);
```

Comment:
In the Python version, I think we execute different code paths based on the value of `options.inplace()`:
```python
    return (_VF.alpha_dropout_(input, p, training)
            if inplace
            else _VF.alpha_dropout(input, p, training))
```
It would be awesome to do the same in C++ version as well (we might need to change the type of `input` from `const Tensor&` to `Tensor` in order to achieve this) :D

Reply:
In `Functions.h` I see two declarations for `alpha_dropout`
``` cpp
static inline Tensor & alpha_dropout_(Tensor & self, double p, bool train) {...}
static inline Tensor alpha_dropout(const Tensor & self, double p, bool train) {...}
```
Guessing we have to use the first one for in-place operation?